github-actions/unified-status-check/src/main.ts

```diff
@@ -0,0 +1,202 @@
+      ...core.getMultilineInput('ignored', {trimWhitespace: true}),
+    ];
+    /** Status matchers which must match at least one of the current statuses . */
+    const required = core.getMultilineInput('required', {trimWhitespace: true});
+    /** The pull request triggering the event */
+    const pullRequest = (
+      await github.graphql<typeof PR_SCHEMA>(query(PR_SCHEMA).toString(), {
+        ...context.issue,
+      })
+    ).repository.pullRequest;
+    /** The status check rollup results. */
+    const statusChecks = pullRequest.commits.nodes[0].commit.statusCheckRollup;
+
+    const setStatus = async (
+      state: 'pending' | 'error' | 'failure' | 'success',
+      description?: string,
+    ) => {
+      await github.repos.createCommitStatus({
+        ...context.repo,
+        sha: pullRequest.commits.nodes[0].commit.oid,
+        context: unifiedStatusCheckName,
+        state,
+        description,
+      });
+      return;
+    };
+
+    /** If no status checks are present, or if the pull request is in a draft state the unified status is in a pending state. */
+    if (!statusChecks || pullRequest.isDraft) {
+      await setStatus('pending', 'Unified Status checks have not yet started');
+      return;
+    }
+
+    const statuses: {state: StatusState; name: string}[] = statusChecks.contexts.nodes
+      .map((checkOrStatus) => {
+        if (checkOrStatus.__typename === 'StatusContext') {
+          return {
+            state: checkOrStatus.state,
+            name: checkOrStatus.context,
+          };
+        }
+        if (checkOrStatus.__typename === 'CheckRun') {
+          return {
+            state: checkOrStatus.conclusion
+              ? checkConclusionStateToStatusStateMap.get(checkOrStatus.conclusion)!
+              : 'PENDING',
+            name: checkOrStatus.name || 'Unknown Check Run',
+          };
+        }
+        throw Error('CheckOrStatus was not found to be a check or a status');
+      })
+      .filter(({name}) => !ignored.includes(name));
+
+    const missedStatuses = required.filter(
+      (matcher) => !statuses.some(({name}) => name.match(matcher)),
+    );
+    if (missedStatuses.length > 0) {
+      await setStatus(
+        'pending',
+        `Waiting for missing required status: ${missedStatuses.join(', ')}`,
+      );
+      return;
+    }
+
+    const counts = statuses.reduce(
+      (count, {state}) => {
+        if (isPassingState(state)) {
+          count.passing += 1;
+        }
+        if (isPendingState(state)) {
+          count.pending += 1;
+        }
+        if (isFailingState(state)) {
+          count.failing += 1;
+        }
+        return count;
+      },
+      {passing: 0, failing: 0, pending: 0},
+    );
+
+    if (counts.failing > 0) {
+      await setStatus('failure', `${counts.failing} expected status(es) failing`);
+      return;
+    }
+
+    if (counts.pending > 0) {
+      await setStatus('pending', 'Expected statuses are still pending');
```

Comment:
Expected is ambiguous IMO because they are not declared as "required". Maybe, `Other statuses are still pending`?